### DISCUSSION TITLE
setup classroom-monitor-bot

### DISCUSSION THREAD

**Discussion by ItsMeIshan** [WeMakeDevs/classroom-monitor-bot#12] [General]
hey, @kaiwalyakoparkar I want to test the bot on my server so that I can understand how it works. I took reference to your blog but for further queries, I am creating this discussion. I hope it adds more value to the project.

**Comment by ItsMeIshan** [comment]
should I create a new bot application on https://discord.com/developers/applications/  to run the code of the bot or is there any other way to test it out?

**Reply by kaiwalyakoparkar** [reply]
You can do that

**Comment by jayesh-srivastava** [comment]
Hey @ItsMeIshan yeah, wanted to know about how to set up the project.

**Comment by ItsMeIshan** [comment]
to set up the project I took help from the [blog](https://dev.to/kaiwalyakoparkar/creating-your-own-discord-bot-using-js-2ep0) created by @kaiwalyakoparkar . The steps are pretty clear in it and after cloning the repo we just need to invite bot to a test server(a server that you can create for your own for testing the bot) and just setup the ```.env``` file with the bot token that you will generate by creating a application [here](https://discord.com/developers/applications/)

**Reply by Amrutha1101** [reply]
After initializing git (i.e. git init) I should copy all the files of this(Classroom-monitor-bot) and paste it in my folder to check if the changes i've made is working or not, Isn't it?

**Reply by jayesh-srivastava** [reply]
Hey I followed the steps in the blog and landed on this but when I am trying to add a server, there's nothing in the list. What should I do?
![Screenshot (745)](https://user-images.githubusercontent.com/49337716/126895867-a2200709-f3d8-4a9d-8206-abfa14fdec81.png)

**Reply by kaiwalyakoparkar** [reply]
In this case create a server on discord. You can name it anything like test or something

**Reply by kaiwalyakoparkar** [reply]
> After initializing git (i.e. git init) I should copy all the files of this(Classroom-monitor-bot) and paste it in my folder to check if the changes i've made is working or not, Isn't it?

Sorry, you don't need to do that step instead clone the repo git will get initialized automatically.

**Reply by kaiwalyakoparkar** [reply]
Do like clone the repo (skip the git init part of the blog) and follow further steps @Amrutha1101

**Reply by Amrutha1101** [reply]
For the BOT_TOKEN should it be enclosed in quotes?

**Reply by Amrutha1101** [reply]
![Screenshot (180)](https://user-images.githubusercontent.com/63070698/126897080-e0f4dbb7-6d7d-43b3-9288-2d88be1b4d63.png)
I got this error @kaiwalyakoparkar can you help?

**Reply by kaiwalyakoparkar** [reply]
Sure @Amrutha1101 if you followed the blog I specified this thing. You will have to copy the bot token and create a `.env` file and add to it also no double quotes or anything. Hope that's clear if not keep sending in your problems I am happy to help :slightly_smiling_face:

**Reply by Amrutha1101** [reply]
I mistook .env-sample with that my bad. it's working fine will send ss

**Reply by jayesh-srivastava** [reply]
Hey @kaiwalyakoparkar I have created a test server and added the bot in it. But I am unable to see its messages, like it did not respond to cc!help

**Reply by Amrutha1101** [reply]
try cm!help

**Reply by jayesh-srivastava** [reply]
Nope, no help

**Reply by kaiwalyakoparkar** [reply]
@jayesh-srivastava Did you clone the repo, and run `node index.js` ?

**Reply by jayesh-srivastava** [reply]
@kaiwalyakoparkar yeah I cloned it but did not run index.js, will do that

**Reply by kaiwalyakoparkar** [reply]
yes that's your local server

**Reply by rockingrohit9639** [reply]
> @kaiwalyakoparkar yeah I cloned it but did not run index.js, will do that

make sure you install all the dependencies using `npm install` or `yarn install` before running `node index.js`
